src/main/java/org/swisspush/redisques/metrics/MetricsCollector.java

```diff
@@ -37,14 +43,18 @@ public class MetricsCollector {
 
     private final AtomicLong activeQueuesCount = new AtomicLong(0);
     private final AtomicLong maxQueueSize = new AtomicLong(0);
+    private final AtomicLong queueStateReadyCount = new AtomicLong(0);
+    private final AtomicLong queueStateConsumingCount = new AtomicLong(0);
+    private final RedisQues redisQues;
 
     public MetricsCollector(Vertx vertx, String uid, String redisquesAddress,
-                            String identifier, MeterRegistry meterRegistry, Lock lock, long metricCollectIntervalSec) {
+                            String identifier, MeterRegistry meterRegistry, Lock lock, long metricCollectIntervalSec, RedisQues redisQues) {
```

Comment:
Adding `RedisQues` in `MetricsCollector`leads to a circular dependency which should be avoided. Please refactor!